protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -38,13 +38,16 @@
 import com.google.protobuf.InvalidProtocolBufferException;
 import com.google.protobuf.MessageLite;
 import com.google.protobuf.Parser;
+import com.google.protobuf.UnsafeByteOperations;
 
 import io.grpc.ExperimentalApi;
 import io.grpc.Metadata;
 import io.grpc.MethodDescriptor.Marshaller;
 import io.grpc.Status;
+import io.grpc.internal.ReadableBuffer;
```

Comment:
I don't understand. You don't get the byte array out, just like you don't get the ReadableBuffer out of the InputStream. Instead, you copy. Instead of using `buffer.readBytes(ByteBuffer)` use `buffer.read(byte[])`.